jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ImageParameters.java

```diff
@@ -70,22 +71,42 @@ public void setCredHelper(String credHelper) {
 
   @Nested
   @Optional
-  AuthConfiguration getAuth() {
+  AuthParameters getAuth() {
     return auth;
   }
 
-  public void auth(Action<? super AuthConfiguration> action) {
+  public void auth(Action<? super AuthParameters> action) {
     action.execute(auth);
   }
 
-  /** Converts the {@link ImageParameters} to an {@link Authorization}. */
+  /**
+   * Converts the {@link ImageParameters} to an {@link Authorization}.
+   *
+   * @param logger the {@link JibLogger} used to print warnings
+   * @param imageProperty the image configuration's name (i.e. "from" or "to")
+   * @return the {@link Authorization}, or null if the username and password aren't both configured
+   */
   @Internal
   @Nullable
-  Authorization getImageAuthorization() {
+  Authorization getImageAuthorization(JibLogger logger, String imageProperty) {
     if (auth.getUsername() == null || auth.getPassword() == null) {
```

Comment:
I'm tempted to also warn when these are `.trim().isEmpty()` (although in such cases, we won't return null). WDYT?

Reply:
Hmm, I'm not sure about usernames, but I could imagine someone making their password all whitespace.

Reply:
Alright, maybe just `.isEmpty()`? I think it'll be fine because we don't error but just warn.